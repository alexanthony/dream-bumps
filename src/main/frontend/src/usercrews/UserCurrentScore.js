import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import {Card, CardTitle, CardText} from 'material-ui/Card';
import { Table, TableBody, TableRow, TableRowColumn } from 'material-ui/Table';

import {getUserCrewRanking} from '../rootReducer';
import {fetchUserCrewRanking} from './actions';

class UserCurrentScore extends Component {
  componentDidMount() {
    this.props.fetchUserCrewRanking(this.props.userId, this.props.sex);
  }

  render() {
    const ranking = this.props.ranking || {};
    return (
      <Card>
        <CardTitle title="Current Score" />
        <CardText>
          <Table
            selectable={false}
          >
            <TableBody
              displayRowCheckbox={false}
            >
              <TableRow>
                <TableRowColumn>Bumps</TableRowColumn>
                <TableRowColumn>{ranking.bumps}</TableRowColumn>
              </TableRow>
              <TableRow>
                <TableRowColumn>Cash</TableRowColumn>
                <TableRowColumn>{ranking.cash}</TableRowColumn>
              </TableRow>
              <TableRow>
                <TableRowColumn>Crew Value</TableRowColumn>
                <TableRowColumn>{ranking.value}</TableRowColumn>
              </TableRow>
              <TableRow>
                <TableRowColumn>Dividends</TableRowColumn>
                <TableRowColumn>{ranking.dividends}</TableRowColumn>
              </TableRow>
              <TableRow>
                <TableRowColumn>Score</TableRowColumn>
                <TableRowColumn>{(ranking.cash + ranking.value).toFixed(2)}</TableRowColumn>
              </TableRow>
            </TableBody>
          </Table>
        </CardText>
      </Card>
    );
  }
}

UserCurrentScore.propTypes = {
  ranking: PropTypes.shape({
    bumps: PropTypes.number,
    cash: PropTypes.number,
    value: PropTypes.number,
    dividends: PropTypes.number
  }),
  fetchUserCrewRanking: PropTypes.func,
  userId: PropTypes.number,
  sex: PropTypes.string
};

export const mapStateToProps = (state, ownProps) => ({
  ranking: getUserCrewRanking(state)(ownProps.userId, ownProps.sex)
});

export {UserCurrentScore};

export default connect(mapStateToProps, {fetchUserCrewRanking})(UserCurrentScore);
