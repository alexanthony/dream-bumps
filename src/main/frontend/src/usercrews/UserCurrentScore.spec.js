import React from 'react';
import { shallow } from 'enzyme';
import { UserCurrentScore } from './UserCurrentScore';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const fetchUserCrewRanking = jest.fn();
  const component = shallow(
    <UserCurrentScore
      fetchUserCrewRanking={fetchUserCrewRanking}
      userId={2}
      sex="female"
      ranking={{bumps: 3, cash: 123.4, value: 1823.43, dividends: 234.23}}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  // We don't actually mount, so these don't get called...
  // expect(fetchUsers).toHaveBeenCalled();
  // expect(fetchUserCrewRankings).toHaveBeenCalled()
});
