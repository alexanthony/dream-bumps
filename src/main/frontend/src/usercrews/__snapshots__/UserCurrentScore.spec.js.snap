exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="Current Score" />
  <CardText>
    <Table
      allRowsSelected={false}
      fixedFooter={true}
      fixedHeader={true}
      height="inherit"
      multiSelectable={false}
      selectable={false}>
      <TableBody
        allRowsSelected={false}
        deselectOnClickaway={true}
        displayRowCheckbox={false}
        multiSelectable={false}
        preScanRows={true}
        selectable={true}
        style={Object {}}>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Bumps
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            3
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Cash
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            123.4
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Crew Value
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            1823.43
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Dividends
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            234.23
          </TableRowColumn>
        </TableRow>
        <TableRow
          displayBorder={true}
          hoverable={false}
          hovered={false}
          selectable={true}
          selected={false}
          striped={false}>
          <TableRowColumn
            hoverable={false}>
            Score
          </TableRowColumn>
          <TableRowColumn
            hoverable={false}>
            1946.83
          </TableRowColumn>
        </TableRow>
      </TableBody>
    </Table>
  </CardText>
</Card>
`;
